nft_ingester/src/processors/transaction_based/bubblegum_updates_processor.rs

```diff
@@ -185,10 +155,31 @@ impl BubblegumTxProcessor {
         result[..4].copy_from_slice(&bytes);
         result
     }
+    pub fn parse_transaction_info_from_fb(
+        data: BufferedTransaction,
+        transaction_parser: Arc<FlatbufferMapper>,
+    ) -> Result<TransactionInfo, IngesterError> {
+        let seen_at = Utc::now();
 
+        let mut transaction_info_bytes = data.transaction.clone();
+
+        if data.map_flatbuffer {
+            let tx_update =
+                    utils::flatbuffer::transaction_info_generated::transaction_info::root_as_transaction_info(
+                        &data.transaction,
+                    ).unwrap();
+            transaction_info_bytes =
+                transaction_parser.map_tx_fb_bytes(tx_update, seen_at).unwrap();
+        }
+        let transaction_info =
+            plerkle_serialization::root_as_transaction_info(transaction_info_bytes.as_slice())
+                .unwrap();
```

Comment:
Are all the `unwraps()` guaranteed to be successful? If so, could you please briefly describe why in a comment?

Reply:
this is code that was copied from another function that i'm not the author of